includes/class-wcsr-resource.php

```diff
@@ -408,4 +428,68 @@ public function set_status( $new_status ) {
 			'to'   => $new_status,
 		);
 	}
+
+	/**
+	 * Calculates the exact active days ratio by looking at the subscription billing cycle and the from_timestamp.
+	 *
+	 * Example case:
+	 * For a monthly subscription, the active_days_ratio for a resource with 61 days_active in a 61 day period previously would result in 1.
+	 * This meant that the cost for 2 month was only 1 * line_item_total. If the subscription's billing period is monthly, the ratio should be 2 so that the cost of the resource for 61 days is 2x the line item total.
+	 *
+	 * To help with the calculations, using the from_timestamp allows us to calculate accurate days in the billing periods that are months and years. Without the from_timestamp there's no way to figure out whether a month counted in the days_in_period was 28,29 (feb in leap years),30 or 31 days. Accounting for the off by one or 2 extra days in Feb was not possible
+	 *
+	 * Using the total days active and the amount days in the period, we can calculate a ratio between 0 and 1 for days active in the period. Then we need multiple that value by the number of billing periods.
+	 *
+	 * @since 1.1.0
+	 * @param int $from_timestamp
+	 * @param int $days_in_period
+	 * @param int $days_active
+	 * @param string $billing_period
+	 * @param int    $billing_interval
+	 * @return float
+	 */
+	public function get_active_days_ratio( $from_timestamp, $days_in_period, $days_active, $billing_period, $billing_interval ) {
```

Comment:
Once the above comment about `get_days_in_period()` not belonging on instances of resources is taken into consideration, it also applies to this method, as the only occasion it uses `$this` is for `$this->get_days_in_period()`.